domoticz: init at 2020.2

nuxeh opened the pull request:
###### Motivation for this change

Add Domoticz home automation server.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nuxeh committed:
maintainers: add edcragg

nuxeh committed:
domoticz: init at 2020.2

nuxeh committed:
domoticz: add module

nuxeh commented:
I have ironed out a few issues brought up by @ofborg, but pending review I think this is ready to go!

nuxeh commented:
Outputs are now in `$out/bin` and `$out/share/domoticz`. I found an undocumented command line parameter allowing the binary to find its static files.

nuxeh commented:
@ofborg eval

nuxeh commented:
Realised hashes keep changing, probably due to use of `deepClone`, I'll rework this.

nuxeh commented:
@ofborg eval

ryantm commented:
@nuxeh I'm going to convert this to a draft, since there is no indication you are done. Feel free to move it out of draft when it is ready.

nuxeh commented:
@ryantm it's ready, thoroughly tested, and has been waiting almost 6 months for review

ryantm commented on nixos/modules/services/misc/domoticz.nix:
```diff
@@ -0,0 +1,89 @@
+    services.domoticz = {
+      enable = mkEnableOption pkgDesc;
+
+      user = mkOption {
```
Can we use systemd dynamic users instead of specifying the user/group? http://0pointer.net/blog/dynamic-users-with-systemd.html

ryantm replied:
https://github.com/nuxeh/nixpkgs/pull/1/files made a PR against your branch that changes it to use StateDirectory/DynamicUser check it out!

nuxeh replied:
Interesting, hadn't heard about this, and previously thought having a configuration option for state dir was a bit clunky.

But thanks, merged.

ryantm commented on pkgs/servers/domoticz/default.nix:
```diff
@@ -0,0 +1,103 @@
+    '';
+    maintainers = with maintainers; [ edcragg ];
+    homepage = "https://www.domoticz.com/";
+    license = licenses.gpl3;
```
The gpl license variants without the `Plus` or `Only` suffix are deprecated, because they do not specify the license well enough. Use `Plus` if the project has "or later" sections in the actual source code files (not the license). Otherwise, use the `Only` suffix.
 
See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

nuxeh replied:
Addressed in 89e5a7f

ryantm commented:
Result of `nixpkgs-review pr 86404` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
<br>- tests.nixos-functions.nixos-test
</details>
<details>
  <summary>1 package built:</summary>
<br>- domoticz
</details>

ryantm commented:
Used [nixos-shell](https://github.com/Mic92/nixos-shell) to test that the service starts up.

ryantm committed:
nixos/domoticz: use DynamicUser and StateDirectory

nuxeh committed:
nixos/domoticz: update to use GPL3 plus license

Per review comments on #86404 and [1].

[1] https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

nuxeh commented:
Have tested with the new dynamic user changes, and runs as expected.

nuxeh commented:
Thanks for reviewing @ryantm, much appreciated!
